### PR TITLE
MAINT: flake8 *.pyx files

### LINKED ISSUE #12995: PEP: fix up *.pyx
we don't by default check the .pyx files, we should. So this issue will cover fixing & updating the ``setup.cfg``, need to add: ``filename: *.py, *.pyx``

### PR DESCRIPTION
**Pull request by gfyoung**
closes #12995 

flake8-ed *.pyx files and fixed errors.

Removed the E226 check because that inhibits pointers (e.g. char*). In addition, the check is not even universally accepted in Python.

### CODE REVIEW

**Comment by jreback** [comment]
not sure what u mean E226 is not universally accepted

pls turn it back on

**Comment by jreback** [comment]
ok, forget about E226, its not so easy to remove.

can you add E128 back to the ``.pxi.in`` as well.

File: ci/lint.sh [hunk 1]
```diff
@@ -20,10 +20,10 @@ if [ "$LINT" ]; then
     echo "Linting *.py DONE"
 
     echo "Linting *.pyx"
-    for path in 'window.pyx' "src/join.pyx"
+    for path in 'algos.pyx' 'hashtable.pyx' 'index.pyx' 'lib.pyx' 'parser.pyx' 'tslib.pyx' 'window.pyx' 'src' 'io' 'msgpack'
     do
         echo "linting -> pandas/$path"
-        flake8 pandas/$path --filename '*.pyx' --select=E501,E302,E203,E226,E111,E114,E221,E303,E128,E231,E126
+        flake8 pandas/$path --filename '*.pyx' --select=E501,E302,E203,E111,E114,E221,E303,E128,E231,E126
```
Comment (jreback) [inline ci/lint.sh]: change this to not use a loop, IOW ``flake8 pandas --filename '*.pyx' --select=.....`` will catch all ``.pyx`` (even if we add/move them)

**Comment by gfyoung** [comment]
@jreback : I'm actually surprised that all of these `flake8` tests passed.  In any case, do you think we can postpone the `pxi.in` changes until a separate PR?

**Comment by jreback** [comment]
no this needs fixing here as u edited the .pyx for algos and join which are auto generated and will revert when the pxi.in are changed

**Comment by gfyoung** [comment]
@jreback : Ah, fair enough.  Will fix then.

**Comment by gfyoung** [comment]
@jreback : On second look, what do you mean that `algos.pyx` is auto-generated?  I believe you mean `algos*.pxi` are auto-generated.  Those are not the same files.

**Comment by jreback** [comment]
right the .pxi are included at cythonization time
so if u only edited .pyx then ok

**Comment by gfyoung** [comment]
@jreback : Nice!  You got Appveyor to work and now block this PR :smile: :angry:

**Comment by jreback** [comment]
oh that's great!

and I stole from numpy a line in appveyor.yml
it cancels the previous if u push again :)
unlike Travis

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**